src/main/java/hudson/plugins/active_directory/ActiveDirectoryUserDetail.java

```diff
@@ -177,6 +177,13 @@ public static long getSerialVersionUID() {
      * Gets the corresponding {@link hudson.model.User} object.
      */
     public hudson.model.User getJenkinsUser() {
+        try { // TODO 1.651.2+ remove reflection
+            return (hudson.model.User) hudson.model.User.class.getMethod("getById", String.class, boolean.class).invoke(getUsername(), true);
+        } catch (NoSuchMethodException x) {
+            // fine, older baseline
+        } catch (Exception e) { // unexpected
+            LOGGER.log(Level.WARNING, String.format("There was a problem obtaining the Jenkins user %s by Id", getUsername()), e);
+        }
```

Comment:
@andresrc Are you sure it is necessary? `getById` should not throw any exception. So if `.invoke` fails you will just go tot the catch of `Exception` to record it. Is it not fine?